### ISSUE TITLE
Diff/commit search returns results outside of time frame

### ISSUE THREAD

**Issue by gabtorre** [sourcegraph/sourcegraph#44132]
- **Sourcegraph version:** 4.1.0 

#### Steps to reproduce:

1. Run a diff/commit search with an _after_ filter
`repo:sourcegraph/sourcegraph type:diff after:"october 25 2022" before:"october 28 2022"`
https://sourcegraph.com/search?q=context:global+repo:sourcegraph/sourcegraph+type:diff+after:%22october+25+2022%22+before:%22october+28+2022%22&patternType=standard

#### Expected behavior: 
Results after 10-25-2022 are returned.

#### Actual behavior:
Results from 10-25-2022 and 10-24-2022 are returned.

<img width="1156" alt="Screen Shot 2022-11-08 at 5 07 45 PM" src="https://user-images.githubusercontent.com/69164745/200711250-9703de1a-c045-4daa-8e64-fbe49261d24e.png">

If would like immediate help on this, please email support@sourcegraph.com (you can still create the issue, but there are not [SLAs](https://about.sourcegraph.com/support/) on issues like there are for support requests).

**Comment by keegancsmith** [comment]
I wonder if this PR resolves the issue, since after is based on a different date than what we display. https://github.com/sourcegraph/sourcegraph/pull/44134

**Comment by camdencheek** [comment]
Yep! this should be fixed by a combination of https://github.com/sourcegraph/sourcegraph/pull/44172 and https://github.com/sourcegraph/sourcegraph/pull/44134#event-7776903752

**Comment by limitedmage** [comment]
Can we update the documentation (autocomplete tips, references panel, docs pages, etc) about the `before` and `after` filters to call out that they work based on UTC?

**Comment by camdencheek** [comment]
On it: https://github.com/sourcegraph/sourcegraph/pull/44216/files
